### ISSUE TITLE
Action creator "initialize" with flag "keepDirty" not working as expected

### ISSUE THREAD

**Issue by Darking360** [erikras/redux-form#2952]
### Are you submitting a **bug report** or a **feature request**?
Bug report

### What is the current behavior?
Greetings, I'm currently working with version 6.7.0, and the bug or issue that I'm presenting is related to the action creator "initialize", the docs says that we should use it as: 

`initialize(form:String, data:Object, [keepDirty:boolean], [options:{keepDirty:boolean, keepSubmitSucceeded:boolean}])`

Now, my form keeps like an email, password and some other info, but when I use "initialize" with keepDirty flag true in order to initialize some countries and states on the form, all the previous data goes away. I tried to use it as:

`this.props.initialize('registerForm',{country_id: 1, state_id: 1}, true)`

Nothing happens, even the data pass to the function doesn't appear on the form, but when I use it as:

`this.props.initialize({country_id: 1, state_id: 1})`

It works, but I cannot set the flag keepDirty.

### What is the expected behavior?
The action creator should keep the data if the "keepDirty" flag is on.

Maybe I'm losing something, so excuse my english and this is the first issue that I create, I used the issue tracker before to find an issue like this but nothing related with the keepDirty flag I think. Thank you and have a nice day.

**Comment by pkbnaveen** [comment]
Is this related to #2723 ?

**Comment by chelzwa** [comment]
This is actually just because the bound action creator provided as a prop to your form doesn't accept options or keepDirty, it only accepts the data object.

I would suggest setting enableReinitialize and keepDirtyOnReinitialize on the form itself, like:
`export default reduxForm({
  form: 'form-name'
  enableReinitialize: true,
  keepDirtyOnReinitialize: true
})(FormComponent);`

Here's the documentation for the function you're calling, showing that it only accepts one parameter (data): http://redux-form.com/6.0.2/docs/api/Props.md/
"**initialize(data:Object) : Function**
Initializes the form data to the given values. All dirty and pristine state will be determined by comparing the current data with these initialized values. This is a bound action creator, so it returns nothing."

**Comment by gustavohenke** [comment]
Unfortunately, that's not the case, @chelzwa.
I checked the code and this is outdated docs on the `initialize`prop arguments.

**Comment by sandkampderek** [comment]
I'm not sure if this is the cause of the OP's problem, but I ran into a similar problem and found a workaround. There is a strange block of code here in the reducer: https://github.com/erikras/redux-form/blob/master/src/createReducer.js#L322

I don't understand why that is there, but it loops through each key in the initialValues that you're trying to set, and if the previous initialValue for that key is undefined, it will go ahead and overwrite your form values with the new initialValues, ignoring the fact that keepDirty is definitely true at that point. I solved this problem by looping through the previous initialValues myself, and if any are undefined, I don't send that key when dispatching initialize().
